### ISSUE TITLE
Support Streaming Audio Data via recognizeUsingWebSocket

### ISSUE THREAD

**Issue by repjarms** [watson-developer-cloud/swift-sdk#1000]
### When you open an issue for a feature request, please add as much detail as possible:

Currently, the interface exposed in `SpeechToTextV1/SpeechToText+Recognize.swift` only leaves a `SpeechToTextSession` alive for the time that it takes to transcribe a `Data` blob.

We should add support to send smaller chunks of data in realtime as a part of one session, to support streaming audio applications that are not driven via the microphone.

**Comment by digitallysavvy** [comment]
@repjarms is this feature still on the roadmap?

**Comment by repjarms** [comment]
@digitallysavvy Yes the plan now is to get it out in a feature level release next week. While I am working on it, can you please provide some more details about the type of interface you are looking for.

For example,

Are you expecting to send an array of bytes, or configure AVAudioSession in a particular way? Any information about how this feature will be used will help me make sure I am delivering something that will address your needs.

**Comment by digitallysavvy** [comment]
@repjarms Great news!

The Agora SDK provides the audio as a Data object that contains only audio data. If we could pass a reference to the audio objects. When we spoke about it, you had mentioned that The SDK already used the Data object but didn't expose it to the developer so I think all we needed was a way to pass Data object directly.

**Comment by zontan** [comment]
@repjarms We're expecting to send an array of bytes. Agora's SDK passes a raw audio buffer, which I'm converting into a `Data` object.  The existing API processes what I have correctly, minus the ability to stream continuously.

**Comment by digitallysavvy** [comment]
@repjarms when @zontan was testing, he was able to send the Data in bursts but it was not a continuous stream (socket) in the way the WatsonMic has its implementation.

**Comment by digitallysavvy** [comment]
@repjarms I realize there probably isn't an update on this but leaving a comment as to keep the issue from closing.

**Comment by mediumTaj** [comment]
Still active

**Comment by digitallysavvy** [comment]
@mediumTaj can we keep this active?

**Comment by digitallysavvy** [comment]
is there still a chance to get access to the mic buffer?
